### PR TITLE
add command to run our benchmarks

### PR DESCRIPTION
**Pull request by RalfJung**
This is quite ad-hoc but better than nothing IMO.
I have also deleted the old `benches` folder. Some of these tests have UB :joy: and the rest doesn't seem very useful to benchmark the things that are slow about Miri today.

Cc @saethlin

### CODE REVIEW

File: bench-cargo-miri/serde2/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "cargo-miri-test"
+version = "0.1.0"
+authors = ["Oliver Schneider <git-spam-no-reply9815368754983@oli-obk.de>"]
+edition = "2018"
+
+[dependencies]
+serde = { version = "*", features = ["derive"] }
+serde_json = "*"
```
Comment (saethlin) [inline bench-cargo-miri/serde2/Cargo.toml:9]: We have lockfiles check in, but I feel like `*` is not what we want here

Comment (RalfJung) [reply bench-cargo-miri/serde2/Cargo.toml:9]: :shrug: this is a literal copy of the serde1 test. But yeah I'll put in something less silly.

**Comment by saethlin** [comment]
I think this is a good start, and it is much better than what I've been doing by hand.

I think we will want a way to report memory usage, but that's a bit of a different beast.

**Comment by RalfJung** [comment]
What's missing is ensuring on CI that these things actually all successfully `cargo run`.

**Comment by RalfJung** [comment]
Oh wow, this is outrageously slow. Maybe we shouldn't run this with stacked borrows cache consistency checking enabled...

**Pull request merged**